bindings/pyroot_experimental/PyROOT/src/TDirectoryPyz.cxx

```diff
@@ -0,0 +1,100 @@
+// Author: Danilo Piparo, Stefan Wunsch CERN  08/2018
+// Original PyROOT code by Wim Lavrijsen, LBL
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "CPyCppyy.h"
+#include "CPPInstance.h"
+#include "CallContext.h"
+#include "PyROOTPythonize.h"
+#include "TPython.h"
+#include "Utility.h"
+
+#include "PyzCppHelpers.hxx"
+
+#include "TClass.h"
+#include "TDirectory.h"
+
+#include "Python.h"
+
+using namespace CPyCppyy;
+
+PyObject *TDirectoryGetObject(CPPInstance *self, PyObject *args)
+{
+   PyObject *name = nullptr;
+   CPPInstance *ptr = nullptr;
+   if (!PyArg_ParseTuple(args, const_cast<char *>("O!O!:TDirectory::GetObject"), &CPyCppyy_PyUnicode_Type, &name,
+                         &CPPInstance_Type, &ptr))
+      return nullptr;
+
+   auto dir = (TDirectory *)OP2TCLASS(self)->DynamicCast(TDirectory::Class(), self->GetObject());
+
+   if (!dir) {
+      PyErr_SetString(PyExc_TypeError,
+                      "TDirectory::GetObject must be called with a TDirectory instance as first argument");
+      return nullptr;
+   }
+
+   void *address = dir->GetObjectChecked(CPyCppyy_PyUnicode_AsString(name), OP2TCLASS(ptr));
+   if (address) {
+      ptr->Set(address);
+
+      Py_INCREF(Py_None);
+      return Py_None;
+   }
+
+   PyErr_Format(PyExc_LookupError, "no such object, \"%s\"", CPyCppyy_PyUnicode_AsString(name));
+   return nullptr;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Type-safe version of TDirectory::WriteObjectAny, which is a template for
```

Comment:
Can we add something more to what this pythonization does, besides the inherited commented from Wim?

Reply:
We can elaborate, clearly, maybe with some text about what WriteObjectAny actually does w/o duplicating too much its doc?

Reply:
Sure, the rational here is that since now we are making the effort of porting these pythonizations one by one, it is the moment to try to understand what they actually do and document it, so that if in the future we need to write a document/tutorial with all the pythonizations available, we already did all the work.